src/ucp/core/ucp_context.c

```diff
@@ -559,6 +559,10 @@ static ucp_tl_alias_t ucp_tl_aliases[] = {
 };
 
 
+/* List of transports that can be used as an auxiliary transport. */
+const char *ucp_context_tl_aux[] = {"ud_verbs", "ud_mlx5", "ugni_udt"};
```

Comment:
why do we need this? In theory we can declare any transport to be `aux`

Reply:
We need this list to not disable the transport from this list as auxiliary transport if the transport was mentioned in the deny list. E.g.:
`UCX_TLS=^ud` should only disable `ud_verbs`, and `ud_mlx5` as a _real_ transport. The output of list of transport should be:
```

Reply:
why is it specific to the transport name, rather than `aux` specificator?

Reply:
We can enable any transport as auxiliary using _allow_ list, e.g. `UCX_TLS=any_transport:aux,any_alias:aux`.
However, we need the list to completely disable transports from this list only if they mentioned in _deny_ list in the following way: `UCX_TLS=^transport_from_the_list:aux,transport_from_the_list`, or `UCX_TLS=^alias_containing_transport_from_the_list:aux,alias_containing_transport_from_the_list`, or `UCX_TLS=^alias_containing_transport_from_the_list:aux,transport_from_the_list`, or `UCX_TLS=^transport_from_the_list:aux,alias_containing_transport_from_the_list`.

Reply:
why not avoiding to disable any transport with `aux` if it is not specified explicitly in the deny list?
with the newly added array the code is error prone - if new aux transport is added to our alias names array it should also be reflected in this array (which is not really obvious)